### ISSUE TITLE
Zip Error: Invalid command arguments (Option 'x' requires value)

### ISSUE THREAD

**Issue by tanvirheer** [lambci/docker-lambda#94]
Looking at the https://hub.docker.com/r/lambci/lambda/ website, there is an example on how to customize your Dockerfile.

So, I have everything working and now i'm trying to run my dockerfile on lambda, however when I run this command below:

Command:

 `CMD cat .lambdaignore | xargs zip -9qyr lambda.zip . -x && aws \
     lambda update-function-code --function-name mylambda --zip-file 
     fileb://lambda.zip `

I get this error: 

```
cat: .lambdaignore: No such file or directory

Error: zip error: Invalid command arguments (option 'x' (exclude files matching patterns) requires a value)
```

Now, is this due to .lambdaignore not existing? Because I checked and it isn't in the docker container when I ssh'd into the container. I created a .lambdaignore and then ran the docker image, it returned no error but it also did nothing.

What is supposed to be in that .lambdaignore?

**Basically, my end-goal is just to push my container onto the cloud and have it run on AWS Lambda.**

**Comment by mhart** [comment]
That's just an example – if you wanted to have files that you didn't want to include in the zip file for example, then you could use an ignore file like that – it's not necessary at all, just a useful pattern

**Comment by mhart** [comment]
When you say "push my container... and have it run on AWS Lambda". Do you mean push a docker container? You can't run docker containers on AWS Lambda.

**Comment by tanvirheer** [comment]
@mhart Wording was off. Basically, the DockerFile contains everything that it needs and I have my script.py file that has the lambda handler function. I tested this function and all the dependences within the docker container when and my program runs. The next step was this program to run on AWS Lambda, I assumed the last line of code in the dockerFile zips the contents up and then when I run the commands below, I could have the contents run on AWS Lambda and have an output.

```
# docker build -t mylambda .
# docker run --rm -e AWS_ACCESS_KEY_ID -e AWS_SECRET_ACCESS_KEY mylambda
```

**Comment by tanvirheer** [comment]
@mhart That makes sense, I don't think I need a .lambdaignore file then.

Regarding the upload, when I do `# docker run --rm -e AWS_ACCESS_KEY_ID -e AWS_SECRET_ACCESS_KEY mylambda` , I don't get any output or message saying it's being uploaded to AWS Lambda.  Therefore, I wasn't sure if it's doing anything at all or it's zipping it up and uploading to AWS Lambda, usually I terminated the process at this point but i'll leave it more than 15 minutes this time to see what happens.

Regarding the dockerFile, all I'm doing is installing pip packages inside the container. The pip packages contain the program that I want to run. Within the dockerFile, I am also manually installing gcc-8.0.1, maybe that is why this is taking so long? The reason for this was, I was getting a GLIB error, saying the file i'm trying to run is looking for different version, and after updating the gcc, it worked fine. 

So is it taking a long time since the only zipping of contents docker is doing must be the gcc-8.0.1 files, because rest are pip3 packages that are just being called from pypi.

Here is my dockerFile if needed:

```
  1 FROM lambci/lambda:build-python3.6
  2 ENV AWS_DEFAULT_REGION us-east-1
  3 
  4 RUN yum install -y wget
  5  #GCC-8.0.1 Installation (Lines 6-18).
  6 RUN cd ~
  7 RUN wget https://ftpmirror.gnu.org/gcc/gcc-8.1.0/gcc-8.1.0.tar.gz
  8 RUN tar xf gcc-8.1.0.tar.gz
  9 RUN cd gcc-8.1.0
 10 WORKDIR gcc-8.1.0/
 11 RUN contrib/download_prerequisites
 12 
 13 RUN cd ~
 14 RUN mkdir build && cd build
 15 RUN ../gcc-8.1.0/configure -v --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x    86_64-linux-gnu --prefix=/usr/local/gcc-8.1 --enable-checking=release --enable-languages=    c,c++,fortran --disable-multilib --program-suffix=-8.1
 16 
 17 RUN make -j 8
 18 RUN make install
 19 
 20 ENV PATH=/usr/local/gcc-8.1/bin:$PATH
 21 ENV LD_LIBRARY_PATH=/usr/local/gcc-8.1/lib64:$LD_LIBRARY_PATH
 22 
 23 RUN pip3 install pybind11
 24 RUN pip3 install lgraph
 25 ADD script.py /
 26 
 27 CMD zip -qyr lambda.zip . && aws \
 28     lambda update-function-code --function-name mylambda --zip-file fileb://lambda.zip
```

and all I'm simply doing is building this docker file and running these two commands
```
# docker build -t mylambda .
# docker run --rm -e AWS_ACCESS_KEY_ID -e AWS_SECRET_ACCESS_KEY mylambda
```

The issue is, I don't see this doing anything on my terminal. There is no confirmation that it got uploaded to lambda, it just doesn't do anything.

**Comment by mhart** [comment]
Hang on – the zipping is so that you can then upload it to AWS Lambda if you want.

To run the function, you don't use a `build-` container – you use one of the normal, non-build containers, and mount the directory containing your (non-zipped) code in /var/task. There are some examples of how to do this in the README

**Comment by tanvirheer** [comment]
Right, I already ran this and it works 100%. I want to now upload it to AWS Lambda, that is the current issue. When I run the docker run command with my aws key access ID, I get output: 

```
/bin/sh: line 1:     8 Killed                  aws lambda update-function-code --function-name mylambda --zip-file fileb://lambda.zip
```

**Comment by mhart** [comment]
Hmmm, not sure what the problem is there – pity it doesn't give you more information.

You could copy the zipfile from the docker container and upload it to AWS locally – and see if there's some sort of error there?

You could use the `docker cp` command for this: https://docs.docker.com/engine/reference/commandline/cp/

Or you could redirect the output from your container command, eg:

```
CMD zip -qyr - .
```

And then
```
docker run --rm mylambda > lambda.zip
aws lambda update-function-code --function-name mylambda --zip-file fileb://lambda.zip
```

**Comment by tanvirheer** [comment]
I could try that. I just ran `docker run --rm mylambda > lambda.zip` and have it in my local machine now zipped up in path: `Users/Tnet/dockerTest/lambda.zip` 

Regarding `aws lambda update-function-code --function-name mylambda --zip-file fileb://lambda.zip` this line, the `fileb://lambda.zip` should be `fileb://Users/Tnet/dockerTest/lambda.zip` correct? For some reason, it is not liking that and is saying 'no such file or directory'

**Comment by mhart** [comment]
Closing this as it's not an issue with docker-lambda
